subsys/bluetooth/audio/ascs.c

```diff
@@ -770,53 +770,30 @@ static uint8_t ascs_attr_cb(const struct bt_gatt_attr *attr, uint16_t handle,
 	return BT_GATT_ITER_CONTINUE;
 }
 
-struct bt_audio_iso *ascs_new_audio_iso(const struct bt_audio_stream *stream,
-					struct bt_ascs *ascs,
-					uint8_t cig_id,
-					uint8_t cis_id)
+static struct bt_audio_iso *audio_iso_get_or_new(struct bt_ascs *ascs, uint8_t cig_id,
+						 uint8_t cis_id)
 {
-	const enum bt_audio_dir dir = stream->ep->dir;
 	struct bt_audio_iso *free_audio_iso = NULL;
 
-	BT_DBG("stream %p dir %u ascs %p", stream, dir, ascs);
+	BT_DBG("ascs %p cig_id 0x%02x cis_id 0x%02x", ascs, cis_id, cis_id);
 
-	/* Look through the audio_isos and find a free one, or one for the same
-	 * ACL connection where the opposite direction is what we want, and is
-	 * free.
-	 *
-	 * The returned audio_iso is either an unused one, or one where @p
-	 * stream can be used in the opposite direction.
-	 */
 	for (size_t i = 0; i < ARRAY_SIZE(ascs->isos); i++) {
 		struct bt_audio_iso *audio_iso = &ascs->isos[i];
-		const struct bt_conn *acl_conn = stream->conn;
-		const struct bt_audio_stream *paired_stream;
-		const struct bt_audio_stream *tmp_stream;
-
-		switch (dir) {
-		case BT_AUDIO_DIR_SINK:
-			tmp_stream = audio_iso->sink_stream;
-			paired_stream = audio_iso->source_stream;
-			break;
-		case BT_AUDIO_DIR_SOURCE:
-			tmp_stream = audio_iso->source_stream;
-			paired_stream = audio_iso->sink_stream;
-			break;
-		default:
-			return NULL;
-		}
+		const struct bt_audio_ep *ep;
 
-		if (tmp_stream == NULL) {
-			if (free_audio_iso == NULL) {
-				free_audio_iso = audio_iso;
-			}
-		} else if (paired_stream != NULL &&
-			   paired_stream->ep->cig_id == cig_id &&
-			   paired_stream->ep->cis_id == cis_id &&
-			   paired_stream->ep->stream->conn == acl_conn) {
-			BT_DBG("Use existing audio_iso %p together with stream %p",
-			       audio_iso, paired_stream);
+		if (audio_iso->sink_stream == NULL && audio_iso->source_stream == NULL) {
+			free_audio_iso = audio_iso;
+			continue;
+		} else if (audio_iso->sink_stream && audio_iso->sink_stream->ep) {
+			ep = audio_iso->sink_stream->ep;
+		} else if (audio_iso->source_stream && audio_iso->source_stream->ep) {
+			ep = audio_iso->source_stream->ep;
+		} else {
+			/* XXX: Stream not associated with endpoint. Should we assert??? */
```

Comment:
Probably not a bad idea

Reply:
can this be somehow triggered by bogus peer device?  if so I'd make it LOG_ERROR/WARN()

Reply:
No, this would be an internal error.